src/test/test.rs

```diff
@@ -82,6 +88,52 @@ mod tests {
         }
     }
 
+    #[test]
+    // Backend must populate ctr_dns_servers via custom
+    // DNS servers for container from container entry and
+    // network dns servers as well.
+    fn test_backend_network_scoped_custom_dns_server() {
+        match config::parse_configs("src/test/config/network_scoped_custom_dns") {
+            Ok((backend, _, _)) => {
+                // Should contain custom DNS server 8.8.8.8
+                let mut dns_server = backend
+                    .ctr_dns_server
+                    .get(&IpAddr::V4(Ipv4Addr::new(10, 88, 0, 2)));
+                let mut expected_dns_server = IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1));
+                assert_eq!(dns_server.unwrap().clone().unwrap()[1], expected_dns_server);
+                expected_dns_server = IpAddr::V6(Ipv6Addr::new(0, 0, 0, 0, 0, 0, 0, 2));
+                assert_eq!(dns_server.unwrap().clone().unwrap()[2], expected_dns_server);
+                expected_dns_server = IpAddr::V4(Ipv4Addr::new(8, 8, 8, 8));
+                assert_eq!(dns_server.unwrap().clone().unwrap()[0], expected_dns_server);
+
+                // Should contain custom DNS servers 3.3.3.3 and 1.1.1.1
+                dns_server = backend
+                    .ctr_dns_server
+                    .get(&IpAddr::V4(Ipv4Addr::new(10, 88, 0, 5)));
+                expected_dns_server = IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1));
+                assert_eq!(dns_server.unwrap().clone().unwrap()[3], expected_dns_server);
+                expected_dns_server = IpAddr::V6(Ipv6Addr::new(0, 0, 0, 0, 0, 0, 0, 2));
+                assert_eq!(dns_server.unwrap().clone().unwrap()[4], expected_dns_server);
+                expected_dns_server = IpAddr::V4(Ipv4Addr::new(3, 3, 3, 3));
+                assert_eq!(dns_server.unwrap().clone().unwrap()[0], expected_dns_server);
+                expected_dns_server = IpAddr::V4(Ipv4Addr::new(1, 1, 1, 1));
+                assert_eq!(dns_server.unwrap().clone().unwrap()[1], expected_dns_server);
+                expected_dns_server = IpAddr::V6(Ipv6Addr::new(0, 0, 0, 0, 0, 0, 0, 1));
+                assert_eq!(dns_server.unwrap().clone().unwrap()[2], expected_dns_server);
+
+                // Shoudld not contain any DNS server other than network's dns server
+                dns_server = backend
+                    .ctr_dns_server
+                    .get(&IpAddr::V4(Ipv4Addr::new(10, 88, 0, 3)));
+                expected_dns_server = IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1));
+                assert_eq!(dns_server.unwrap().clone().unwrap()[0], expected_dns_server);
+                expected_dns_server = IpAddr::V6(Ipv6Addr::new(0, 0, 0, 0, 0, 0, 0, 2));
+                assert_eq!(dns_server.unwrap().clone().unwrap()[1], expected_dns_server);
+            }
```

Comment:
Please rewrite this an a way that is more readable. Some for loop instead of the big amount of duplication would help a lot.

Reply:
Adding loop SGTM